utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -60,31 +56,93 @@ pub struct TryRuntimeCmd {
 		default_value = "Interpreted"
 	)]
 	pub wasm_method: WasmExecutionMethod,
+
+	/// The state to use to run the migration.
+	#[structopt(subcommand)]
+	pub state: State,
 }
 
 /// The state to use for a migration dry-run.
-#[derive(Debug)]
+#[derive(Debug, structopt::StructOpt)]
 pub enum State {
-	/// A snapshot. Inner value is a file path.
-	Snap(String),
+	/// Use a snapshot as state to run the migration.
+	Snap {
+		#[structopt(flatten)]
+		cache_params: CacheParams,
+	},
+
+	/// Use a live chain to run the migration.
+	Live {
+		/// An optional cache file to WRITE to. Not cached if set to `None`.
+		#[structopt(short, long)]
+		cache_file: Option<CacheParams>,
+
+		/// The block hash at which to connect.
+		/// Will be latest finalized head if not provided.
+		#[structopt(short, long, multiple = false, parse(try_from_str = parse_hash))]
+		block_at: Option<String>,
+
+		/// The modules to scrape. If empty, entire chain state will be scraped.
+		#[structopt(short, long, require_delimiter = true)]
+		modules: Option<Vec<String>>,
+
+		/// The url to connect to.
+		#[structopt(default_value = "http://localhost:9933", parse(try_from_str = parse_url))]
+		url: String,
+	},
+}
 
-	/// A live chain. Inner value is the HTTP uri.
-	Live(String),
+fn parse_hash(block_number: &str) -> Result<String, String> {
+	let block_number = if block_number.starts_with("0x") {
+		&block_number[2..]
+	} else {
+		block_number
+	};
+
+	if let Some(pos) = block_number.chars().position(|c| !c.is_ascii_hexdigit()) {
+		Err(format!(
+			"Expected block hash, found illegal hex character at position: {}",
+			2 + pos,
+		))
+	} else {
+		Ok(block_number.into())
+	}
+}
+
+fn parse_url(s: &str) -> Result<String, &'static str> {
+	if s.starts_with("http://") {
```

Comment:
What about `https://`? I think that should be supported too.

Reply:
I don't know if our nodes even expose https endpoints, do they? 

Either way, this is rooted in my old code. Back in the days, the `ws/wss` endpoints where not capable of delivering the whole state, and only HTTP could.

Reply:
> I don't know if our nodes even expose https endpoints, do they?

I bet most production deployments stick a TLS proxy in front of their nodes. And we will support https directly in `jsonrpsee`. :)